modtools/create-unit.lua

```diff
@@ -566,6 +587,7 @@ if args.name then
 else
   local unit = df.unit.find(unitId)
   unit.name.has_name = false
+  unit.name.first_name = "" -- remove the first name altogether. gets rid of units having number names.
```

Comment:
Oh, sorry about that. I'm not sure what's up with the notifications - I managed to comment in a way that I don't usually use, and my comment didn't show up on my end for a few minutes, so maybe that's related.

I'll look into this again as soon as I can (in the middle of r1 right now), and see about adding separate handling for arena mode if necessary.